### ISSUE TITLE
Tool quick usage (using a tool while holding a button and then auto switching back) bug

### ISSUE THREAD

**Issue by Equbuxu** [PixiEditor/PixiEditor#294]
Right now it is determined based on how long you hold the button down for but actually, it should switch back if you start using the tool before releasing the button

**Comment by flabbet** [comment]
I'm not sure about that. This doesn't seem very convienient. If I tap a tool shortcut I expect it to switch to it, not wait for me to draw anything

**Comment by Equbuxu** [comment]
> I'm not sure about that. This doesn't seem very convienient. If I tap a tool shortcut I expect it to switch to it, not wait for me to draw anything

I don't think you get it. Right now quick usage is determined by repeat events, it works like this:

Press E -> Eraser is selected -> Release right away (no repeat events) -> Eraser stays selected (you've just switched your tool)
Press E -> Erase for a bit (repeat events are firing) -> Release -> Old tool is selected (quick usage)
Press E -> Erase very quickly (no repeat events cause you're faster than them) -> Release -> Eraser stays selected (you wanted to "quick use" the eraser but you was a bit too quick and it switched instead)

Quick usage should instead be determined by whether you've started using the tool before or after releasing the E key:

Press E -> Eraser is selected -> Release without erasing -> Eraser stays selected (you've just switched your tool)
Press E -> Erase -> Release -> Old tool is selected (quick usage)
Press E -> Erase very quickly -> Release -> Old tool is selected (quick usage actually worked, cause it knows you erased something)
